### ISSUE TITLE
[Question, possible BUG] Output of _TCNModule is a function of input_chunk_length, not output_chunk_length

### ISSUE THREAD

**Issue by DanielBergThomsen** [unit8co/darts#1965]
Hello!

I've been working with the _TCNModule of the library, and only noticed now that the size of the output is the following shape:
`x = x.view(batch_size, self.input_chunk_length, self.target_size, self.nr_params)`
This strikes me as a bit odd; why would we want to produce target predictions for the past? In that case the model ought to just output the targets it has access to directly.

Best regards,
Daniel

**Comment by DanielBergThomsen** [comment]
To be more specific, this is the part of the code I am referencing:
https://github.com/unit8co/darts/blob/6096968537da2a6d45591692329fdbdea48b8829/darts/models/forecasting/tcn_model.py#L234-L248

The reason this is working right now is that `input_chunk_length` is passed as the `length` parameter in `TCNModel`:
https://github.com/unit8co/darts/blob/6096968537da2a6d45591692329fdbdea48b8829/darts/models/forecasting/tcn_model.py#L478-L493

But this means that losses are calculated based on entries not related to `output_chunk_length`.

**Comment by DanielBergThomsen** [comment]
After reading the documentation I noticed the following comment, which seems to indicate that this is intentional behavior:
https://github.com/unit8co/darts/blob/6096968537da2a6d45591692329fdbdea48b8829/darts/models/forecasting/tcn_model.py#L177-L183

It seems different from other implementations of TCNs (https://github.com/locuslab/TCN for example).

The reason this could be a problem is that this indeed does result in the training losses being based on the targets of previous time steps. As a user, I don't particularly care for this I just want the model to be able to predict the future `output_chunk_length` number of targets accurately.

I would just like confirmation whether I've missed something, if I didn't miss anything I would be down to create a PR to try and change it.

**Comment by dennisbader** [comment]
Hi @DanielBergThomsen, and sorry for the late response.

Thanks for opening this issue. I think you are right, the loss shouldn't be computed on the target values that were fed to the model as input.

I raised PR #2006 to address this.
